src/web/rustdoc.rs

```diff
@@ -320,15 +320,23 @@ pub fn rustdoc_html_server_handler(req: &mut Request) -> IronResult<Response> {
 
     content.full = file_content;
 
-    let latest_version = crate_details.latest_version().to_owned();
+    let latest_release = crate_details.latest_release();
+    let latest_version = latest_release.version.to_owned();
     let is_latest_version = latest_version == version;
 
-    let path_in_latest = if !is_latest_version {
+    let latest_path = if is_latest_version {
+        format!("/{}/{}", name, latest_version)
+    } else if latest_release.build_status {
         let mut latest_path = req_path.clone();
         latest_path[2] = &latest_version;
-        path_for_version(&latest_path, &crate_details.doc_targets, &conn)
+        format!(
+            "/{}/{}/{}",
+            name,
+            latest_version,
+            path_for_version(&latest_path, &crate_details.doc_targets, &conn)
+        )
     } else {
-        Default::default()
+        format!("/crate/{}/{}", name, latest_version)
```

Comment:
Just so I understand - the reason for the change is so that the link to the latest doesn't give a 404 if the build doesn't succeed? As far as I know this should have worked the same way as you have here before, just with an extra redirect.

Reply:
Yep, this is just removing one extra redirect step, and making the behaviour match what the test thought it was testing 😄